language/move-vm/runtime/src/interpreter.rs

```diff
@@ -101,15 +102,45 @@ impl Interpreter {
         extensions: &mut NativeContextExtensions,
         loader: &Loader,
     ) -> VMResult<Vec<Value>> {
-        Interpreter {
+        let mut interpreter = Interpreter {
             operand_stack: Stack::new(),
             call_stack: CallStack::new(),
             paranoid_type_checks: loader.vm_config().paranoid_type_checks,
             runtime_limits_config: loader.vm_config().runtime_limits_config.clone(),
+        };
+        if function.is_native() {
+            for arg in args {
+                interpreter
+                    .operand_stack
+                    .push(arg)
+                    .map_err(|e| e.finish(Location::Undefined))?;
+            }
+            let resolver = function.get_resolver(loader);
+            let return_values = interpreter
+                .call_native_return_values(
+                    &resolver,
+                    data_store,
+                    gas_meter,
+                    extensions,
+                    function.clone(),
+                    &ty_args,
+                )
+                .map_err(|e| match function.module_id() {
+                    Some(id) => e
+                        .at_code_offset(function.index(), 0)
+                        .finish(Location::Module(id.clone())),
+                    None => PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                        .with_message(
+                            "Unexpected native function not located in a module".to_owned(),
+                        )
+                        .finish(Location::Undefined),
+                })?;
+            Ok(return_values.into_iter().collect())
```

Comment:
Is there some equivalent to the paranoid checks we do in the other code path that should go here?